lyngk_board.js

```diff
@@ -0,0 +1,210 @@
+    }
+    var boardString = document.getElementById('boardstring').innerHTML;
+    translateBoardString(boardString, context);
+    populateClaimedCanvases();
+  }
+
+  function drawLine(x1, y1, x2, y2, ctx) {
+    ctx.beginPath();
+    ctx.moveTo(x1, y1);
+    ctx.lineTo(x2, y2);
+    ctx.stroke();
+  }
+
+  function drawGamePiece(xcenter, ycenter, color, ctx) {
+    ctx.beginPath();
+    ctx.arc(xcenter, ycenter, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[color];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function drawUnclaimedPiece(color, piecePosition) {
+    var board =document.getElementById("unclaimedCanvas");
+    var ctx=board.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(40, piecePosition * 62, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[letterToColor[color]];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function drawClaimedPiece(color, boardNumber, piecePosition) {
+    boardLabel = '';
+    switch(boardNumber) {
+      case 1:
+        boardLabel = "player1canvas";
+        break;
+      case 2:
+        boardLabel = "player2canvas";
+        break;
+    }
+    var board =document.getElementById(boardLabel);
+    var ctx=board.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(piecePosition * 29, 25, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[letterToColor[color]];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function populateClaimedCanvases() {
+    var unclaimedBoard = document.getElementById("unclaimedCanvas");
+    var unclaimedctx = unclaimedBoard.getContext("2d");
+    unclaimedctx.clearRect(0, 0, unclaimedBoard.width, unclaimedBoard.height);
+    var claimedBoard1 = document.getElementById("player1canvas");
+    var player1ctx = claimedBoard1.getContext("2d");
+    player1ctx.clearRect(0, 0, claimedBoard1.width, claimedBoard1.height);
+    var claimedBoard2 = document.getElementById("player2canvas");
+    var player2ctx = claimedBoard2.getContext("2d");
+    player2ctx.clearRect(0, 0, claimedBoard2.width, claimedBoard2.height);
+    var stringArray1 = document.getElementById('player1claimed').innerHTML.split('');
+    var stringArray2 = document.getElementById('player2claimed').innerHTML.split('');
+    var player1count = 1;
+    var player2count = 1;
+    var unclaimedCount = 1;
+    for(var i = 0; i < colorArray.length; i++) {
+      if (stringArray1.indexOf(colorArray[i]) != -1) {
+        drawClaimedPiece(colorArray[i], 1, player1count);
+        player1count += 1;
+      }
+      else if (stringArray2.indexOf(colorArray[i]) != -1) {
+        drawClaimedPiece(colorArray[i], 2, player2count);
+        player2count += 1;
+      }
+      else {
+        drawUnclaimedPiece(colorArray[i], unclaimedCount);
+        unclaimedCount += 1;
+      }
+    }
+  }
+
+  function stackGamePiece(xcenter, ycenter, color, stackCount, ctx) {

```

Comment:
Try to name functions what they do.  Perhaps in this case, "drawStackedGamePiece"?